blockstack/blockstackd.py

```diff
@@ -3579,24 +3764,51 @@ def run_blockstackd():
         json_data = {}
         json_data['balances'] = stx_balances
         json_data['vesting'] = vesting
+        json_data['namespaces'] = namespaces
+        json_data['names'] = names
 
         # write the json output file
-        print "Writing migration data to {}".format(args.output_path)
-        with open(args.output_path, 'w') as json_out:
-            json.dump(json_data, json_out, separators=(',', ':'))
+        print_status("Writing on-chain migration data to json...")
+        json_output_path = os.path.join(output_dir, 'chainstate.json')
+        json_hash_output_path = json_output_path + '.sha256'
+        with open(json_output_path, 'w') as json_out:
+            json.dump(json_data, json_out, separators=(',', ':'), check_circular=False, ensure_ascii=False)
```

Comment:
Is there a reason `ensure_ascii=False` is necessary?  Like, are there non-ASCII characters in the DB somewhere?

Reply:
IIUC all chars in the DB are ascii. `ensure_ascii` defaults to True, so I set as a minor optimization that prevents python from running the non-ascii escape routine on all the strings. Is this incorrect, should I remove it?

Reply:
They should all be ASCII, yes.  But, the key word here is _should_ -- I think the node should crash if it finds a name that isn't ASCII, since that would indicate a bug we'll want to correct.  Is the performance hit substantial?

Reply:
When the debugger isn't attached, the perf difference is negligible, so I can remove.